board/freescale/mx6ullevk/mx6ullevk.c

```diff
@@ -27,38 +27,41 @@
 
 DECLARE_GLOBAL_DATA_PTR;
 
-#define SKL_BOARD_TYPE_PCB195 0
-#define SKL_BOARD_TYPE_PCB213 1
-#define SKL_BOARD_TYPE_PCB217 2
-#define SKL_BOARD_TYPE_PCB277 3
-
-#define GPIO2_IO18	IMX_GPIO_NR(2, 18)
-#define GPIO2_IO19	IMX_GPIO_NR(2, 19)
-#define GPIO2_IO20	IMX_GPIO_NR(2, 20)
-#define GPIO2_IO21	IMX_GPIO_NR(2, 21)
+#ifndef CONFIG_SPL_BUILD
 
-static unsigned int get_siklu_board_id()
+/* See comment on MTDPARTS_DEFAULT in include/configs/mx6ullevk_siklu.h.
+ * This function is called in cmd/mtdparts.c. */
+void board_mtdparts_default(const char **mtdids, const char **mtdparts)
 {
-	unsigned int id = 0;
-
-	gpio_direction_input(GPIO2_IO18);
-	gpio_direction_input(GPIO2_IO19);
-	gpio_direction_input(GPIO2_IO20);
-	gpio_direction_input(GPIO2_IO21);
-
-	id = id | gpio_get_value(GPIO2_IO21);
-	id = id << 1;
-	id = id | gpio_get_value(GPIO2_IO20);
-	id = id << 1;
-	id = id | gpio_get_value(GPIO2_IO19);
-	id = id << 1;
-	id = id | gpio_get_value(GPIO2_IO18);
-	return id;
+	static canary = 0;
+	static char *siklu_mtdparts;
+	static char ids[24];
```

Comment:
What is the reason for static variables here? Is it to reduce `siklu_get_board_type()` calls?

Reply:
Copied the code structure from siklu_get_board_type() itself to avoid unnecessary assignments and calls to siklu_get_board_type(). Probably doesn't save much.

Reply:
Yes, it is to reduce the calls to siklu_get_board_type() to only one call per uptime.